File: articles/secure-computation-and-privacy-preserving-algorithms/fortran/privacy_preserving_demo.f90
program privacy_preserving_demo
  implicit none
  real :: weights(3)
  integer :: counts(3)
  real :: weighted
  integer :: total

  weights = (/0.42, 0.55, 0.49/)
  counts = (/100, 240, 160/)
  weighted = sum(weights * real(counts))
  total = sum(counts)

  print *, 'federated average weight=', weighted / real(total)
end program privacy_preserving_demo

File: articles/secure-computation-and-privacy-preserving-algorithms/java/PrivacyPreservingDemo.java
public class PrivacyPreservingDemo {
    public static void main(String[] args) {
        double[] weights = {0.42, 0.55, 0.49};
        int[] counts = {100, 240, 160};
        double weighted = 0.0;
        int total = 0;
        for (int i = 0; i < weights.length; i++) {
            weighted += weights[i] * counts[i];
            total += counts[i];
        }
        System.out.printf("federated average weight=%.6f%n", weighted / total);
    }
}
